polars/polars-io/src/parquet/read_impl.rs

```diff
@@ -303,6 +297,35 @@ pub fn read_parquet<R: MmapBytesReader>(
     }
 }
 
+#[allow(clippy::too_many_arguments)]
+pub fn read_parquet<R: MmapBytesReader>(
+    mut reader: R,
+    limit: usize,
+    projection: Option<&[usize]>,
+    schema: &ArrowSchema,
+    metadata: Option<FileMetaData>,
+    predicate: Option<Arc<dyn PhysicalIoExpr>>,
+    parallel: ParallelStrategy,
+    row_count: Option<RowCount>,
+) -> PolarsResult<DataFrame> {
+    let file_metadata = metadata
+        .map(Ok)
+        .unwrap_or_else(|| read::read_metadata(&mut reader))?;
+    let reader = ReaderBytes::from(&reader);
+    let bytes = reader.deref();
+    let column_store = mmap::ColumnStore::Local(bytes);
```

Comment:
I don't think `ColumnStore` is a good name. Can't `local` store whole row-groups?

Reply:
@ritchie46  thanks for looking at the PR and I agree it's not a good name. To specifically answer you question, in `local` store we don't actually store anything, it's just a pass through.

MORE CONTEXT (sorry this is a bit long)

We are a bit in a tight spot: 
- when running locally we want to use mmap
- when running remote we want to download only the parts of the file of interest (footer or last 64KB as per one of the comments)
- we currently want the async code to be as self-contained as possible

The ColumnStore is an abstraction that I introduced in an earlier PR to allow us to keep most of the code as-is and specially limit the spread of `async`.  To achieve this this layer provides a level of indirection where the reader uses linear addressing, so the code stays as is. 

For the local use case  the ColumnStore is pass through:
  - it maps the address requested by the reader to the actual address in the file
  - the OS is doing its magic
  -  the code that you are looking at here is a no-op.

For the remote use case: 
- we pre-download the regions of the file that we know will be accessed later on.  
- This allows us to go in async mode for the period of the download and then come back to non-async mode. 
- All the downloaded areas get store in a hashmap. 
- When the reader executes and accesses an address we return the downloaded buffer.

I see a couple of ways forward:
1. allow the reader to run in async mode and actually do the download in the reader when the region of the file is needed
2. keep the current async island but come with a better name for the CloudStore

#1 is a bit of a departure from the current code architecture and you seem to prefer threads to async. Over the long run it may be inevitable though since we need to teach the planner the cost of operations.

If you want to go with #2 I could rename this to ColumMapper or CloudMapper, or some other name that resonates more with you.

Let me know what you think :hugs:

Reply:
Sorry for the delay (really busy). Great explanation. Could we add this as comments on top of the `ColumnStore`. Maybe we call it `CloudMapper` indeed.


I think we should first go for #2. I see some opportunities for some `async` in the streaming engine. But I don't want to overhaul our design and gradually experiment with this.